src/main/kotlin/app/revanced/patches/youtube/misc/downloadactions/HookDownloadActionsPatch.kt

```diff
@@ -0,0 +1,174 @@
+package app.revanced.patches.youtube.misc.downloadactions
+
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
+import app.revanced.patcher.extensions.InstructionExtensions.addInstructionsWithLabels
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.util.smali.ExternalLabel
+import app.revanced.patches.youtube.misc.downloadactions.fingerprints.*
+import app.revanced.patches.youtube.utils.compatibility.Constants
+import app.revanced.patches.youtube.utils.integrations.Constants.INTEGRATIONS_PATH
+import app.revanced.patches.youtube.utils.integrations.Constants.MISC_PATH
+import app.revanced.patches.youtube.utils.mainactivity.MainActivityResolvePatch
+import app.revanced.patches.youtube.utils.resourceid.SharedResourceIdPatch
+import app.revanced.patches.youtube.utils.settings.SettingsPatch
+import app.revanced.util.getReference
+import app.revanced.util.indexOfFirstInstructionOrThrow
+import app.revanced.util.patch.BaseBytecodePatch
+import app.revanced.util.resultOrThrow
+import com.android.tools.smali.dexlib2.Opcode
+import com.android.tools.smali.dexlib2.iface.instruction.OneRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.formats.Instruction35c
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+
+@Suppress("unused")
+object HookDownloadActionsPatch : BaseBytecodePatch(
+    name = "Hook download actions",
+    description = "Adds options to show the download playlist button and hook the download actions.",
+    dependencies = setOf(
+        MainActivityResolvePatch::class,
+        SharedResourceIdPatch::class,
+        SettingsPatch::class
+    ),
+    compatiblePackages = Constants.COMPATIBLE_PACKAGE,
+    fingerprints = setOf(
+        OfflineVideoEndpointFingerprint,
+        PiPPlaybackFingerprint,
+        AccessibilityOfflineButtonSyncFingerprint,
+        DownloadPlaylistButtonOnClickFingerprint
+    )
+) {
+    private const val INTEGRATIONS_DOWNLOAD_PLAYLIST_BUTTON_CLASS_DESCRIPTOR =
+        "$MISC_PATH/HookDownloadAction;"
+
+    private const val INTEGRATIONS_VIDEO_UTILS_CLASS_DESCRIPTOR =
+        "$INTEGRATIONS_PATH/utils/VideoUtils;"
+
+    override fun execute(context: BytecodeContext) {
+
+        // region Patch for hook download actions
+
+        OfflineVideoEndpointFingerprint.resultOrThrow().mutableMethod.apply {
+            addInstructionsWithLabels(
+                0, """
+                    invoke-static/range {p3 .. p3}, $INTEGRATIONS_VIDEO_UTILS_CLASS_DESCRIPTOR->inAppDownloadButtonOnClick(Ljava/lang/String;)Z
```

Comment:
The hook for the video download button is handled in the `VideoUtils` class, but the hook for the playlist download button is handled in the `HookDownloadAction` class.

If you want to separate the `Hook download actions` patch from the `Overlay buttons` patch, it would be more appropriate to change it so that it is handled in one class.